vault/external_tests/raft/raft_autopilot_test.go

```diff
@@ -340,3 +340,54 @@ func TestRaft_Autopilot_Stabilization_Delay(t *testing.T) {
 	}
 	require.Equal(t, state.Voters, []string{"core-0", "core-1", "core-2"})
 }
+
+func TestRaft_AutoPilot_Peersets_Equivalent(t *testing.T) {
+	cluster := raftCluster(t, &RaftClusterOpts{
+		InmemCluster:         true,
+		EnableAutopilot:      true,
+		DisableFollowerJoins: true,
+	})
+	defer cluster.Cleanup()
+	testhelpers.WaitForActiveNode(t, cluster)
+
+	// Create a very large stabilization time so we can test the state between
+	// joining and promotions
+	client := cluster.Cores[0].Client
+	_, err := client.Logical().Write("sys/storage/raft/autopilot/configuration", map[string]interface{}{
+		"server_stabilization_time": "1h",
+	})
+	require.NoError(t, err)
+
+	joinFunc := func(core *vault.TestClusterCore) {
+		_, err := core.JoinRaftCluster(namespace.RootContext(context.Background()), []*raft.LeaderJoinInfo{
+			{
+				LeaderAPIAddr: client.Address(),
+				TLSConfig:     cluster.Cores[0].TLSConfig,
+				Retry:         true,
```

Comment:
With Retry=true, it looks like JoinRaftCluster spawns a goroutine and always returns nil, so this test may not be deterministic.

Reply:
fixed in https://github.com/hashicorp/vault/pull/11158